src/en/flamecomics/src/eu/kanade/tachiyomi/extension/en/flamecomics/FlameComics.kt

```diff
@@ -4,62 +4,158 @@ import android.graphics.Bitmap
+            ?: return MangasPage(listOf(), false)
+
+        val manga = searchedSeriesData.filter { series ->
+            val titles = json.decodeFromString<List<String>>(series.altTitles) + series.title
+            titles.any { title -> query in removeSpecialCharsregex.replace(title.lowercase(), "") }
+        }.map { seriesData ->
+            SManga.create().apply {
+                title = seriesData.title
+                setUrlWithoutDomain("$baseUrl/series/${seriesData.series_id}")
+                thumbnail_url = "$cdn/series/${seriesData.series_id}/${seriesData.cover}"
             }
-            .mapIndexed { i, el ->
-                if (el.tagName() == "p") {
-                    Page(i, "", el.select("img").attr("abs:src"))
-                } else {
-                    val imageUrls = el.select("img")
-                        .joinToString("|") { it.attr("abs:src") }
-
-                    Page(i, document.location(), imageUrls + COMPOSED_SUFFIX)
-                }
+        }
+        page--
+
+        var lastPage = page * 20 + 20
+        if (lastPage > manga.size) {
+            lastPage = manga.size
+        }
+        if (lastPage < 0) lastPage = 0
+        return MangasPage(manga.subList(page * 20, lastPage), lastPage < manga.size)
+    }
+
+    override fun searchMangaNextPageSelector(): String? = null
+    override fun popularMangaNextPageSelector(): String? = null
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET("$baseUrl/browse?search=$query&page=$page")
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun searchMangaSelector(): String = throw UnsupportedOperationException()
+    override fun popularMangaSelector(): String = "div[class^=SeriesCard_imageContainer] img"
+    override fun latestUpdatesSelector(): String =
+        "div[class^=SeriesCard_chapterImageContainer] img"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.attr("abs:alt").substringAfter("$baseUrl/").replaceFirst("/", "")
+        val url = element.attr("abs:src")
+        thumbnail_url = getRealUrl(url).substringBefore("&")
+        setUrlWithoutDomain(getRealUrl(url).substringBefore("/thumb"))
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga =
+        popularMangaFromElement(element)
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        val seriesData = getJsonData<PageData>(document)?.props?.pageProps?.series
+
+        title = seriesData?.title + ""
+
+        thumbnail_url = "$cdn/series/${seriesData?.series_id}/${seriesData?.cover}"
```

Comment:
Since `seriesData` might be null you should handle this differently.

```suggestion
        seriesData?.let {
            thumbnail_url = "$cdn/series/${it.series_id}/${it.cover}"
        }
```

(although you can wrap more of the surrounding code into this `let` block, maybe use a different name than `it` in that case)